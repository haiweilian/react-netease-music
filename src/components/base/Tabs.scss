@include b(tabs) {
  display: flex;
  justify-content: flex-end;
  list-style: none;

  @include e(item) {
    padding: 12px 0;
    margin: 0 12px;
    font-size: 12px;
    color: #7f7f81;
    white-space: nowrap;
    cursor: pointer;

    @include when(active) {
      color: #d33a31;
    }

    &:hover {
      color: #000;
    }
  }
}
