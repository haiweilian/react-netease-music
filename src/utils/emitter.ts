import mitt from 'mitt'

export default mitt<{
  spin: boolean
}>()
