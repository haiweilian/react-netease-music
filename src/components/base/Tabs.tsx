import './Tabs.scss'
import { useEffect, useState } from 'react'
import type { ISingleTab } from '~/types'

interface TabsProps {
  tabs: ISingleTab[]
  value?: string | number
  onChange: (value: string | number) => void
}

export default function Tabs({ tabs, value, onChange }: TabsProps) {
  const [currentValue, setCurrentValue] = useState<string | number>()
  useEffect(() => {
    setCurrentValue(value)
  }, [value])

  return (
    <ul className="tabs">
      {tabs.map((tab) => (
        <li
          key={tab.value}
          className={tab.value === currentValue ? 'tabs__item is-active' : 'tabs__item'}
          onClick={() => {
            setCurrentValue(tab.value)
            onChange(tab.value)
          }}
        >
          {tab.label}
        </li>
      ))}
    </ul>
  )
}
