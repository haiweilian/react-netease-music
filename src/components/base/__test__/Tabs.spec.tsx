import { render, screen, fireEvent } from '@testing-library/react'
import { describe, expect, test, vi } from 'vitest'
import { localPlaylistTabs } from '~/utils/local'
import Tabs from '../Tabs'

describe('Tabs', () => {
  test('should tabs', () => {
    const onChange = vi.fn()
    const first = localPlaylistTabs[0].value
    const last = localPlaylistTabs[localPlaylistTabs.length - 1].value
    const { container } = render(<Tabs tabs={localPlaylistTabs} value={first} onChange={onChange} />)

    // init
    expect(screen.getByText(first)).toBeTruthy()
    expect(container).toMatchSnapshot()

    // click
    fireEvent.click(screen.getByText(last))
    expect(screen.getByText(last)).toBeTruthy()
    expect(onChange).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })
})
