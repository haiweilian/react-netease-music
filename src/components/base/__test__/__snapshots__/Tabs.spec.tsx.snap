// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should tabs 1`] = `
<div>
  <ul
    class="tabs"
  >
    <li
      class="tabs__item is-active"
    >
      全部
    </li>
    <li
      class="tabs__item"
    >
      欧美
    </li>
    <li
      class="tabs__item"
    >
      华语
    </li>
    <li
      class="tabs__item"
    >
      流行
    </li>
    <li
      class="tabs__item"
    >
      说唱
    </li>
    <li
      class="tabs__item"
    >
      摇滚
    </li>
    <li
      class="tabs__item"
    >
      民谣
    </li>
    <li
      class="tabs__item"
    >
      电子
    </li>
    <li
      class="tabs__item"
    >
      轻音乐
    </li>
    <li
      class="tabs__item"
    >
      ACG
    </li>
    <li
      class="tabs__item"
    >
      怀旧
    </li>
    <li
      class="tabs__item"
    >
      治愈
    </li>
    <li
      class="tabs__item"
    >
      旅行
    </li>
  </ul>
</div>
`;

exports[`Tabs should tabs 2`] = `
<div>
  <ul
    class="tabs"
  >
    <li
      class="tabs__item"
    >
      全部
    </li>
    <li
      class="tabs__item"
    >
      欧美
    </li>
    <li
      class="tabs__item"
    >
      华语
    </li>
    <li
      class="tabs__item"
    >
      流行
    </li>
    <li
      class="tabs__item"
    >
      说唱
    </li>
    <li
      class="tabs__item"
    >
      摇滚
    </li>
    <li
      class="tabs__item"
    >
      民谣
    </li>
    <li
      class="tabs__item"
    >
      电子
    </li>
    <li
      class="tabs__item"
    >
      轻音乐
    </li>
    <li
      class="tabs__item"
    >
      ACG
    </li>
    <li
      class="tabs__item"
    >
      怀旧
    </li>
    <li
      class="tabs__item"
    >
      治愈
    </li>
    <li
      class="tabs__item is-active"
    >
      旅行
    </li>
  </ul>
</div>
`;
